File: backend/src/types/resource.ts
export type Provider = '115';
export type MediaType = 'movie' | 'tv';

export interface ResourceDto {
  id: string;
  title: string;
  provider: Provider;
  mediaType: MediaType;
  rawType: string;
  size: string;
  shareUrl: string;
  extra: Record<string, unknown>;
}

File: backend_py/app/models/resource.py
from typing import Literal

from pydantic import BaseModel, Field

Provider = Literal["115"]
MediaType = Literal["movie", "tv"]


class ResourceDto(BaseModel):
    id: str
    title: str
    provider: Provider
    mediaType: MediaType
    rawType: str
    size: str
    shareUrl: str
    extra: dict[str, object] = Field(default_factory=dict)
